tasks/services/works.js

```diff
@@ -31,31 +31,34 @@ function completeOrderSetupTaskWork (hook) {
 
 function createCastOrderIntentTaskPlan (hook) {
   const taskPlans = hook.app.service('taskPlans')
+  const orders = hook.app.service('orders')
   const relationships = hook.app.service('relationships')
   const taskRecipeId = taskRecipes.castIntent.id
   const completeOrderTaskPlanId = hook.data.taskPlanId
+  let params
 
   return taskPlans.get(completeOrderTaskPlanId)
   .then((completeOrderTaskPlan) => {
-    const groupId = completeOrderTaskPlan.params.consumerAgentId
     const orderId = completeOrderTaskPlan.params.orderId
-    const supplierAgentId = completeOrderTaskPlan.params.supplierAgentId
-    const params = { orderId, consumerAgentId: groupId, supplierAgentId }
-
+    return orders.get(orderId)
+  })
+  .then((order) => {
+    const { id, consumerAgentId, supplierAgentId, adminAgentId } = order
+    params = { orderId: id, consumerAgentId, supplierAgentId, adminAgentId }
```

Comment:
why not pass this down the promise pipeline? it seems real sketchy (and prone to race conditions) to pass it through a global variable.

you could instead do:

```js
.then((order) => {
  const params = {}
  return Promise.all([
    Promise.resolve(params),
    relationships.find({})
  ])
})
.then(([params, relationships]) => {
})
```